docs/modules/library-development/pages/classes.adoc

```diff
@@ -0,0 +1,63 @@
+# Create a Class
+
+## Library Structure
+
+Classes should be placed in the `src` directory. It is recommended that a package be used and named after the library contributing the class.
+
+[TIP]
+====
+Each loaded library's `src` directory contents are synced to a common directory.  Colliding file paths will result in an exception being thrown.
+====
+
+## Serializability
+
+In order to Jenkins pipelines to resume, a design pattern called Continuation Passing Style (CPS) has been implemented in the Pipeline Plugin. CPS works by continuously serializing the current state of the pipeline to disk such that if the pipeline were to be paused or Jenkins restarted, the pipeline could resume where it left off.
+
+Therefore, every variable that's persisted in the pipeline's context must be serializable.
+
+To help ensure that the classes that are contributed by a library are serializable, they should implement the `Serializable` interface.
+
+For example, a `Utility` class coming from an `example` library:
+
+[source, groovy]
+----
+package example
+
+class Utility implements Serializable {}
+----
+
+[NOTE]
+====
+To learn more, check out https://www.jenkins.io/doc/book/pipeline/pipeline-best-practices/#avoiding-notserializableexception[Best Pracies for Avoiding Serializability Exceptions]
+====
+
+## Accessibility
+
+Classes contributed by loaded libraries can be import from the pipeline template, steps of the same library, and steps from other libraries.
+
+## Jenkins DSL Step Resolution
+
+Library classes can not resolve Jenkins pipeline DSL functions such as `sh` or `echo`. A work around for this is to pass the `steps` variable to the class constructor to store on a field or through a method parameter.
```

Comment:
Can we also pass another variables to read the pipeline config or library config?

Reply:
I believe that is an example. you can probably add another setter or constructor to pass more variables or set other fields

Reply:
Yup! The classes will behave more or less like typical groovy classes so you can create whatever method parameters or constructor parameters you need, including the `config` or `pipelineConfig` variables

Reply:
Good suggestion! We'll add it to the docs